src/element/linearElementEditor.ts

```diff
@@ -9,18 +9,17 @@ import { getElementPointsCoords } from "./bounds";
 import { Point, AppState } from "../types";
 import { mutateElement } from "./mutateElement";
 import { SceneHistory } from "../history";
-import { globalSceneState } from "../scene";
 
 export class LinearElementEditor {
-  public elementId: ExcalidrawElement["id"];
+  public element: NonDeleted<ExcalidrawLinearElement>;
```

Comment:
Not a fan of this. I [moved from `element` → `elementId`](https://github.com/excalidraw/excalidraw/pull/1616/commits/77352b884097bb46efd15770408a6ae9493c9e2d) because it manifested in multiple bugs and was very difficult to maintain referential integrity.

Reply:
Yeah, I remember you mentioning this earlier, should I also maintain a global map of `elementId=>element`? Or a map of `ExcalidrawElemenntId => local scene`?

Reply:
That wouldn't work because WeakMaps only work on objects (they must be references to memory). I don't how to handle this case. There are options, but none ideal. Will think on it.

Reply:
Yeah weakmaps will only work on objects but what I meant is Map! 
But as you said this might not be ideal soln.

Reply:
If we use maps, there would be no garbage collection so would lead to memory leaks after long use (many instances, many elements...).

Reply:
How about having global map of `uniqueId => LocalScene` where unique id is `Date.now()` ?
In that case, we will have only one key for one instance, and on unmount we can remove that as well.

Reply:
@dwelle @petehunt  thoughts?

Reply:
Ad your last point, I don't see any benefit to the status quo. The problem is how to get a reference to a store if all you have is an element object at your disposal.

(And I'm not addressing cases where all you have is string element `id`, or not even that.)

Reply:
> Ad your last point, I don't see any benefit to the status quo. The problem is how to get a reference to a store if all you have is an element object at your disposal.
> 
> (And I'm not addressing cases where all you have is string element `id`, or not even that.)

I am not sure If I followed correctly. 
What I meant is If we can have a global map like `{ excalId=>globalscene} ` where `excalId` is a unique Id and since its global so everywhere it can be accessed. 
This `excalID` can have two use cases I think
1. It will store local scene for every instance(since every instance will have its own id)
2. Local storage can also use this `excalID` as if we have multiple excal instances but local storage. has one key then again it will end up sharing the scenes unless local storage save is disabled.
I am not sure if we use `excalElement` as the key in weakMap, will we be able to fix the local storage sharing.